libs/product/src/selectors/configurable-product/configurable-product.selectors.ts

```diff
@@ -57,6 +61,31 @@ const createConfigurableProductSelectors = (): DaffConfigurableProductMemoizedSe
 		}
 	);
 
+	/**
+	 * Selector for the range of discounts for current configuration of the configurable product.
+	 */
+	const selectConfigurableProductDiscountedPrices = createSelector(
+		selectProductEntities,
+		selectConfigurableProductAppliedAttributesEntitiesState,
+		(products, appliedAttributesEntities, props) => {
+			return selectMatchingConfigurableProductVariants.projector(products, appliedAttributesEntities, { id: props.id })
+				.map(variant => variant.discount ? variant.price - variant.discount.amount : variant.price);
+		}
+	);
+
+	/**
+	 * Selector that determines whether any of the variants for the current configuration of the configurable product has a discount.
+	 */
+	const selectConfigurableProductHasDiscount = createSelector(
+		selectProductEntities,
+		selectConfigurableProductAppliedAttributesEntitiesState,
+		(products, appliedAttributesEntities, props) => {
+			return selectMatchingConfigurableProductVariants.projector(products, appliedAttributesEntities, { id: props.id })
+				.filter(variant => variant.discount ? variant.discount.amount > 0 : false)
+				.length > 0;
```

Comment:
This should be implemented with `reduce`.

Reply:
why? this is shorter, right?

Reply:
You're computing a single value that depends on every value in a list. That is a reduce operation so I think its more clear to express it as such.

I'm not sure which implementation would actually contain fewer characters, but a `reduce` implementation is just a single operation instead of 2: `filter` and then a comparison on `length`.

```ts
reduce((variant, acc) => acc || (variant.discount && variant.discount.amount > 0), false)
```

Reply:
ok done